tests/foundry/lemmas.k

```diff
@@ -0,0 +1,113 @@
+requires "../contracts.k"
+
+module TEST-HELPER-LEMMAS
+  imports ASSERTTEST-CONTRACT
+
+  syntax Bytes ::= testFunction() [function]
+  syntax Set ::= testFunction2(Bytes) [function]
+
+  rule testFunction() => AssertTest.test_assert_true() [simplification]
+  rule testFunction() => #binRuntime(AssertTest) [simplification]
+  rule testFunction2(PGM) => #computeValidJumpDests(PGM) [simplification]
+
+endmodule
+
+module SUM-TO-N-INVARIANT
+
+  imports LOOPSTEST-CONTRACT
+
+    rule N xorInt maxUInt256 <Int S => false
+    requires #rangeUInt(256, S +Int ((N *Int (N +Int 1)) /Int 2))
+    andBool #rangeUInt(256, S)
+    andBool #rangeUInt(256, N)
+    [simplification]
+
+    rule (S +Int N) +Int (((N -Int 1) *Int N) /Int 2) <Int pow256 => true
+    requires #rangeUInt(256, S +Int ((N *Int (N +Int 1)) /Int 2))
+    andBool #rangeUInt(256, S)
+    andBool #rangeUInt(256, N)
+    [simplification]
+
+    rule chop(S +Int N) => S +Int N
+    requires #rangeUInt(256, S +Int ((N *Int (N +Int 1)) /Int 2))
+    andBool #rangeUInt(256, S)
+    andBool #rangeUInt(256, N)
+    [simplification]
```

Comment:
This, when turned into a claim, can be proven by KEVM master. Why is it needed as a lemma?
```
    claim [chop-simpl-1]: runLemma ( chop(S +Int N) ) => doneLemma ( S +Int N )
      requires #rangeUInt(256, S +Int ((N *Int (N +Int 1)) /Int 2))
       andBool #rangeUInt(256, S)
       andBool #rangeUInt(256, N)
```
Separately, just having `0 <=Int S andBool 0 <=Int N` should be sufficient, the upper bound on them is not needed as it is established by the other constraint.

Reply:
I removed it.